refinery/tool_manager/models.py

```diff
@@ -62,16 +63,18 @@ class FileRelationship(models.Model):
         (LIST, "list")
     )
     uuid = UUIDField(unique=True, auto=True)
-    name = models.TextField(max_length=100, blank=False,
-                            null=False)
-    value_type = models.CharField(
-        max_length=100, choices=RELATIONSHIP_TYPES, blank=False, null=False)
-    file_relationship = models.ManyToManyField("self", symmetrical=False,
-                                               null=True, blank=True)
+    name = models.TextField(max_length=100)
+    value_type = models.CharField(max_length=100, choices=RELATIONSHIP_TYPES)
+
+    # NOTE: `symmetrical=False` is not very common. It's necessary for the
+    # self-referential M2M below. See: http://bit.ly/2mpPQfT
+    file_relationship = models.ManyToManyField(
+        "self", symmetrical=False, null=True, blank=True)
```

Comment:
We want to provide some asymmetry to the FileRelationship nesting. Without this, the bottom-most FileRelationship could know about the top-most one, and things could get messy. This way `FileRelationships` know about their nested children only.